### PR TITLE
Add new property hidden to formControl fields

### LINKED ISSUE #1659: [Feature] Hidden (dynamic and static) fields in query forms
## Summary
Feature which has allowance for config driven hiding of form controls in query forms

## Motivation
Query forms are driven by configurations, this allows for client agnostic development of plugins.
To keep the config driven architecture, while allowing for further flexibility in visibility of columns this feature is needed.

## Additional Context

We will have a new property called `hidden` in the control configurations.
Example:
```
hidden: true
```
This signifies that the control will be of input type "hidden" with a static value provided in the `initialValue` property of the configuration.

Example:
```
hidden: { path: "actionConfiguration.body[0].x", comparison: "EQUALS", value: "y" }
```

Here we have a structure for the `hidden` property which the client understands. 
`path` this is a json path to the property we're trying to compare.
`comparison` this is the type of comparison operation we need to perform the potential values will be put in an enum.
`enum ComparisonOperations { "EQUALS", "NOT_EQUALS", "GREATER", "LESSER" }` (the actual strings are subject to change and only for reference.
`value` the RHS of this comparison

In the above example, the comparison performed by the client will be `_.get(actionConfiguration, "body[0].x") === "y"`.

_*Note on further abstraction*_
Today this approach allows for one comparison. This doesn't include cases which need more complex comparisons and calculations to come to a conclusion of whether to hide a certain control.

_*Note on logical parity with Property Pane controls*_
PropertyControls don't have to rely on JSON and can be passed in the format `hidden: (props: Value) => boolean` as the configurations come from the respective widgets.

One approach we can take to come to an eventual parity with the PropertyControls, is to have a ControlGenerator, which generates query form controls based on the configuration. This will lead us to eventually use a single `Control` component which generates the controls for both the property pane and query forms.

The underlying similarity between PropertyControls and Query form controls is only the fact that the controls are rendered based on a configuration. If we decide to merge the two controls, we will have to achieve parity w.r.t the config structures as well.

### PR DESCRIPTION
**Pull request by akash-codemonk**
Fixes #1659

### CODE REVIEW

File: app/client/src/components/formControls/utils.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import _ from "lodash";
```
Comment (riodeuno) [inline app/client/src/components/formControls/utils.ts:1]: Let us start tree-shaking lodash. We can `import { get, isBoolean } from "lodash"`

**Review by hetunandu — CHANGES_REQUESTED** [review]
The hidden logic should be handled in the BaseControl and not in each and every control. This is leading to lots of code duplication

File: app/client/src/components/formControls/utils.ts (hunk 1 above)
Comment (riodeuno) [reply app/client/src/components/formControls/utils.ts:1]: Let us add a unit test for this function.
It should to see if it handles scenarios of `undefined` or `null` values in the `hidden` config object properties gracefully.

**Review by riodeuno — CHANGES_REQUESTED** [review]
@akash-codemonk when the value is READ_ONLY, we do not render the `Field` this leads me to believe, that the value may not available when the form is submitted. Could you confirm if the READ_ONLY fields have their original values sent to the servers?

**Comment by akash-codemonk** [comment]
@hetunandu Will create a separate component which will create the control. Similar to https://github.com/appsmithorg/appsmith/blob/release/app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx, this will handle the hidden logic and avoid duplication.

**Comment by akash-codemonk** [comment]
@riodeuno The form values are initialised with the action or datasource values with for e.g `changeQuerySaga`. So yes the original value still remains in the form state when it is submitted even if the field is hidden.

**Review by mohanarpit — APPROVED** [review]

File: app/client/src/components/formControls/BaseControl.tsx [hunk 2]
```diff
@@ -9,12 +9,15 @@ abstract class BaseControl<P extends ControlProps, S = {}> extends Component<
...
 }
 
-export enum ComparisonOperations {
-  EQUALS = "EQUALS",
-  NOT_EQUALS = "NOT_EQUALS",
-  GREATER = "GREATER",
-  LESSER = "LESSER",
-}
+export type ComparisonOperations =
+  | "EQUALS"
```
Comment (riodeuno) [inline app/client/src/components/formControls/BaseControl.tsx:13]: Please elaborate on the reasoning behind moving to string literals from enum.

**Review by riodeuno — COMMENTED** [review]
Looks good to me.

File: app/client/src/components/formControls/BaseControl.tsx (hunk 2 above)
Comment (akash-codemonk) [reply app/client/src/components/formControls/BaseControl.tsx:13]: The json is going to be in a string format.

**Pull request merged**